include/NotificationsManager.php

```diff
@@ -28,109 +32,762 @@ public function __construct() {
+
+		if(dbm::is_result($r))
+			return $r[0]['total'];
+
+		return 0;
+	}
+
+	/**
+	 * @brief Get personal notifications
+	 * 
+	 * @param int|string $seen
+	 *	If 0 only include notifications into the query
+	 *	which aren't marked as "seen"
+	 * @param int $start Start the query at this point
+	 * @param int $limit Maximum number of query results
+	 * 
+	 * @return array with
+	 *	string 'ident' => Notification identifier
+	 *	int 'total' => Total number of available personal notifications
+	 *	array 'notifications' => Personal notifications
+	 */
+	public function personalNotifs($seen = 0, $start = 0, $limit = 80) {
+		$ident = 'personal';
+		$total = $this->personalTotal($seen);
+		$sql_extra .= $this->_personal_sql_extra();
+		$notifs = array();
+
+		if($seen === 0)
+			$sql_seen = " AND `item`.`unseen` = 1 ";
+
+		$r = q("SELECT `item`.`id`,`item`.`parent`, `item`.`verb`, `item`.`author-name`, `item`.`unseen`,
+				`item`.`author-link`, `item`.`author-avatar`, `item`.`created`, `item`.`object` AS `object`, 
+				`pitem`.`author-name` AS `pname`, `pitem`.`author-link` AS `plink`, `pitem`.`guid` AS `pguid`, 
+			FROM `item` INNER JOIN `item` AS `pitem` ON  `pitem`.`id`=`item`.`parent`
+			WHERE `item`.`visible` = 1
+				$sql_extra
+				$sql_seen
+				AND `item`.`deleted` = 0 AND `item`.`uid` = %d AND `item`.`wall` = 0 
+			ORDER BY `item`.`created` DESC LIMIT %d, %d " ,
+				intval(local_user()),
+				intval($start),
+				intval($limit)
+		);
+
+		if(dbm::is_result($r))
+			$notifs = $this->formatNotifs($r, $ident);
+		
+		$arr = array (
+			'notifications' => $notifs,
+			'ident' => $ident,
+			'total' => $total,
+		);
+
+		return $arr;
+	}
+
+	/**
+	 * @brief Total number of home notifications 
+	 * @param int|string $seen
+	 *	If 0 only include notifications into the query
+	 *	which aren't marked as "seen"
+	 * @return int Number of home notifications
+	 */
+	private function homeTotal($seen = 0) {
+		if($seen === 0)
+			$sql_seen = " AND `item`.`unseen` = 1 ";
+
+		$r = q("SELECT COUNT(*) AS `total` FROM `item`
+				WHERE `item`.`visible` = 1 AND
+				 `item`.`deleted` = 0 AND `item`.`uid` = %d AND `item`.`wall` = 1
+				$sql_seen",
 			intval(local_user())
 		);
-    }
+
+		if(dbm::is_result($r))
+			return $r[0]['total'];
+
+		return 0;
+	}
+
+	/**
+	 * @brief Get home notifications
+	 * 
+	 * @param int|string $seen
+	 *	If 0 only include notifications into the query
+	 *	which aren't marked as "seen"
+	 * @param int $start Start the query at this point
+	 * @param int $limit Maximum number of query results
+	 * 
+	 * @return array with
+	 *	string 'ident' => Notification identifier
+	 *	int 'total' => Total number of available home notifications
+	 *	array 'notifications' => Home notifications
+	 */
+	public function homeNotifs($seen = 0, $start = 0, $limit = 80) {
+		$ident = 'home';
+		$total = $this->homeTotal($seen);
+		$notifs = array();
+
+		if($seen === 0)
+			$sql_seen = " AND `item`.`unseen` = 1 ";
```

Comment:
Like above (and at several other places). Please define a default value for "sql_seen".